### ISSUE TITLE
stack new -- AesonException

### ISSUE THREAD

**Issue by gitfoxi** [commercialhaskell/stack#2275]
`stack new hello-world` fails with

```
Could not parse '/home/m/.stack/config.yaml':
AesonException "failed to parse field 'params': when expecting a Text, encountered Number instead"
See http://docs.haskellstack.org/en/stable/yaml_configuration.html.
```

When my `config.yaml` looks like

```
templates:
  params:
    author-email: 415fox@gmail.com
    author-name: michael fox
    category: Application
    copyright: copytright michael fox 2016
    github-username: gitfoxi
    year: 2016
```

Works fine if I change the year line to:

```
    year: "2016"
```

I think it's pretty clear what's going on here, but the documentation at http://docs.haskellstack.org/en/stable/yaml_configuration/ doesn't mention the `year` parameter. I learn about from when I run `stack new` with a blank `config.yaml`. It says:

```
Downloading template "new-template" to create project "hello-world" in hello-world/ ...
The following parameters were needed by the template but not provided: author-email, author-name, category, copyright, github-username, year
You can provide them in /home/m/.stack/config.yaml, like this:
templates:
  params:
    author-email: value
    author-name: value
    category: value
    copyright: value
    github-username: value
    year: value
```

I think it could be fixed by requiring that `year` be a `string` instead of a `value` or by coercing whatever `value` happens to be to a string using Aeson's encode function.

**Comment by mgsloan** [comment]
I'd prefer just special casing "Number" than using the encode function.  Otherwise, more complicated errors could turn into complicated strings, and the yaml accepted.

**Comment by tswelsh** [comment]
I took a quick look at this as another place to dip into the stack codebase.  According to what's in `Stack.New` the year parameter is always set to the current year, unless overridden by an explicit `-p year:2016` on the command line.  A parameter with name `year` in `~/.stack/config.yaml` never gets used according to: https://github.com/commercialhaskell/stack/blob/db81ab360a9511d93187bcc522732a79718372dc/src/Stack/New.hs#L181.  

Is this what you would expected?  Would you ever want to use a non-current year in the config file?

However, even if the value is never used you still see the exception when parsing `config.yaml`.  Was the suggestion to allow all parameters to be `Number`, or just `year`?  The alternative is just documenting what the code does, i.e. all parameter values must be a string.

**Comment by mgsloan** [comment]
@tswelsh Sorry for the delay in responding, I am currently cleaning up my browser window with "TODO" stack issues.

> Is this what you would expected? Would you ever want to use a non-current year in the config file?

Good question.  It seems arbitrary to disallow it, so it's allowed but not recommended.

> Was the suggestion to allow all parameters to be Number, or just year? 

All parameters should be allowed to be Number.  A similar change recently happened in hpack - https://github.com/sol/hpack/commit/8aa7537fedc8d0ac032f8a85606b40c349a3bd03#diff-87ed521fa5387e6fab1b0cdf6786eb63R76

**Comment by tswelsh** [comment]
Can this be closed off as #3701 has been merged?  Sorry should have linked to this in the PR somehow

**Comment by lwm** [comment]
No stress, let's close it, sure!
